checks/check_extra764

```diff
@@ -36,11 +36,14 @@ extra764(){
       fi
 
       # https://aws.amazon.com/premiumsupport/knowledge-center/s3-bucket-policy-for-config-rule/
-      CHECK_BUCKET_STP_POLICY_PRESENT=$(cat $TEMP_STP_POLICY_FILE | jq --arg arn "arn:aws:s3:::${bucket}" '.Statement[]|select(((.Principal|type == "string") and .Principal == "*") and .Action=="s3:*" and (.Resource|type == "array") and (.Resource|map({(.):0})[]|has($arn)) and (.Resource|map({(.):0})[]|has($arn+"/*")) and .Condition.Bool."aws:SecureTransport" == "false")')
-      if [[ $CHECK_BUCKET_STP_POLICY_PRESENT ]]; then
-        textPass "Bucket $bucket has S3 bucket policy to deny requests over insecure transport"
-      else
-        textFail "Bucket $bucket allows requests over insecure transport"
+      # checking if $TEMP_STP_POLICY_FILE is a valid json before converting it to json with jq
+      if jq -e . >/dev/null 2>&1 <<< "$TEMP_STP_POLICY_FILE"; then
+        CHECK_BUCKET_STP_POLICY_PRESENT=$(cat $TEMP_STP_POLICY_FILE | jq --arg arn "arn:aws:s3:::${bucket}" '.Statement[]|select(((.Principal|type == "string") and .Principal == "*") and .Action=="s3:*" and (.Resource|type == "array") and (.Resource|map({(.):0})[]|has($arn)) and (.Resource|map({(.):0})[]|has($arn+"/*")) and .Condition.Bool."aws:SecureTransport" == "false")')
```

Comment:
@orevron AWS generally validates json before accepting it, under what condition are you receiving invalid json? Is it an error condition that is not caught above? Assumign that you've submitted invalid json and AWS has passed if back to you, don't you think it would be better to testInfo rather the silently ignoring it?

Reply:
i got it when tried to delete s3 bucket from console but the resource staied as zombie resource (the resource is not available but still exists)
in this case a string error is coming back from this request..

added else statment with text info